### ISSUE TITLE
File extension - Game options Arguments

### ISSUE THREAD

**Issue by MoonDragon-MD** [lutris/lutris#4602]
Hi, I would like to know how to set the 'Arguments' in a variable manner via the net.lutris.*.desktop launcher
Let me explain in more detail:
my intention is to be able to open a certain file type (specific extension) with a programme installed in Lutris.

Note: 
*= program name
**= program extension *

I have created in 
`$HOME/.local/share/applications/`
a file called
`wine-extension-**.desktop`
with inside

```
[Desktop Entry]
Type=Application
Name=*
MimeType=application/**;
Exec=env LUTRIS_SKIP_INIT=1 lutris lutris:rungameid/4 start /ProgIDOpen **file %f
NoDisplay=true
StartupNotify=true
Icon=lutris_*
```

Then I went to the file and set it to open with the chosen programme.
Now if I double-click on the file it opens the programme but not the file.
So I thought that the problem may be due to the path of gnome3 (/path/file.**) instead of written in windows mode (Z:\path/file.**)
so I modified the file 
$HOME/.local/share/applications/wine-extension-**.desktop
like this

```
[Desktop Entry]
Type=Application
Name=*
MimeType=application/**;
Exec=$HOME/**.sh
NoDisplay=true
StartupNotify=true
Icon=lutris_*
```

In the **.sh file I wrote

```
#!/bin/sh
via=`cat tmp.txt`
echo "$@" >> $HOME/.**tmp.txt
sed -i -e 's,/,\\/,g' $HOME/.**tmp.txt
Exec=env LUTRIS_SKIP_INIT=1 lutris lutris:rungameid/4 Z:$via
rm $HOME/.**tmp.txt
exit
```

Where:
- 4 is the bottle number of the programme installed in Lutris
- Z: is the Ubuntu operating system partition assigned to wine/Lutris 

Now this also only opens the program without the file.
So I would like to know in which way or in which file I can write/change the value of Arguments or if there is another solution
Thank you

**Comment by MoonDragon-MD** [comment]
I found a way to fix it thanks to a user on the Italian Ubuntu forum, for those interested you will find the whole code here: [](https://forum.ubuntu-it.org/viewtopic.php?p=5320476#p5320476)

**Comment by strycore** [comment]
The arguments after the game identifier aren't passed to the game, that's why Lutris is only opening the program.

What is the solution proposed on the Italian forums?

**Comment by danieljohnson2** [comment]
It looks to me like they are editing Lutris's .config .yml files before running Lutris. *Nasty*.

**Comment by MoonDragon-MD** [comment]
> It looks to me like they are editing Lutris's .config .yml files before running Lutris. _Nasty_.

Yes exactly it was the only way at least for now, it works great and on any peripheral/pathway. I don't understand why you find it "nasty"?

**Comment by strycore** [comment]
That's the only way possible to achieve this tbh. And YAML is much simpler than Steam's VDF format and SteamTinkerLaunch does the exact same thing for Steam.

In a weird Mandela effect moment, I was almost sure we had worked on passing extra arguments to the game but apparently we never did.

**Comment by danieljohnson2** [comment]
:confused: Lutris itself does not use sed; it uses PyYAML, doesn't it?

I'm not saying editing VDF files with sed wouldn't be crazier.

**Comment by strycore** [comment]
they could use a small python script using PyYAML as well but keep in mind Linux users have a fetish for doing everything using Unix shell commands :upside_down_face:

**Comment by MoonDragon-MD** [comment]
> Well, I don't speak Italian but I looks like they are using sed to do the edit. YAML is a very complicated format; that's not really an adequate tool to do it safely.

Do you want me to translate the script into English for you?

**Comment by danieljohnson2** [comment]
No, thanks, I don't want to get hot sticky sed all over me. Blech!

**Comment by strycore** [comment]
It's ok, I just wanted what mechanism was at play here. As I said before, I thought we already had worked on this. Hopefully, passing remaining arguments for the game should not be hard. It might require passing `--` between the Lutris arguments and the game arguments.

**Comment by MoonDragon-MD** [comment]
> It's ok, I just wanted what mechanism was at play here. As I said before, I thought we already had worked on this. Hopefully, passing remaining arguments for the game should not be hard. It might require passing `--` between the Lutris arguments and the game arguments.

I think it is not difficult to implement the ability to open a default file extension in Lutris, maybe you can take a look at how Playonlinux does it. Thanks

**Comment by MoonDragon-MD** [comment]
A practical example: 
File.xyz > right click (or automatic assignment) > open with Program 

This takes you to our *.sh with the script to edit the *.yml file, which will then launch the exe installed in Lutris that will ad to read the modified argument i.e. our File.xyz

**Comment by strycore** [comment]
In case it wasn't clear, we're not opting for a solution that involves editing yml files. The solution is extremely simple and trivial and only involves passing command line arguments to the game.

**Comment by superboo07** [comment]
Any updates on a solution that doesn't involve editing lutris config files?
